java-frontend/src/main/java/org/sonar/java/se/MethodBehavior.java

```diff
@@ -55,10 +55,14 @@ public void createYield(ProgramState programState, boolean happyPathYield) {
     }
 
     if (!isConstructor() && !isVoidMethod()) {
-      SymbolicValue resultSV = programState.peekValue();
+      SymbolicValue resultSV = programState.returnValue();
       if (resultSV != null) {
         yield.resultIndex = parameterSymbolicValues.indexOf(resultSV);
         yield.resultConstraint = programState.getConstraint(resultSV);
+      } else if (happyPathYield) {
+        // FIXME A resultSV should always be present for these methods at the end of the execution path,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1134)